<!DOCTYPE html>
<!--[if lt IE 7]>      <html class="no-js lt-ie9 lt-ie8 lt-ie7"> <![endif]-->
<!--[if IE 7]>         <html class="no-js lt-ie9 lt-ie8"> <![endif]-->
<!--[if IE 8]>         <html class="no-js lt-ie9"> <![endif]-->
<!--[if gt IE 8]><!--> <html class="no-js"> <!--<![endif]-->
    <head>
        <meta charset="utf-8">
        <meta http-equiv="X-UA-Compatible" content="IE=edge,chrome=1">
        <title>bitBio - JW</title>
        <meta name="description" content="">
        <meta name="viewport" content="width=device-width">
            <link rel="stylesheet" href="http://wjidea.github.io/theme/css/normalize.css">
        <link href='//fonts.googleapis.com/css?family=Lato' rel='stylesheet' type='text/css'>
        <link href='//fonts.googleapis.com/css?family=Oswald' rel='stylesheet' type='text/css'>
        <link rel="stylesheet" href="http://wjidea.github.io/theme/css/font-awesome.min.css">
        <link rel="stylesheet" href="http://wjidea.github.io/theme/css/main.css">

    <link rel="stylesheet" href="http://wjidea.github.io/theme/css/blog.css">
    <link rel="stylesheet" href="http://wjidea.github.io/theme/css/github.css">
        <link href="http://wjidea.github.io/feeds/all.atom.xml" type="application/atom+xml" rel="alternate" title="bitBio Atom Feed" />
        <script src="http://wjidea.github.io/theme/js/vendor/modernizr-2.6.2.min.js"></script>
    </head>
    <body>
        <!--[if lt IE 7]>
            <p class="chromeframe">You are using an <strong>outdated</strong> browser. Please <a href="http://browsehappy.com/">upgrade your browser</a> or <a href="http://www.google.com/chromeframe/?redirect=true">activate Google Chrome Frame</a> to improve your experience.</p>
        <![endif]-->

        <div id="wrapper">
<header id="sidebar" class="side-shadow">
    <hgroup id="site-header">
        <a id="site-title" href="http://wjidea.github.io"><h3><i class="icon-coffee"></i> bitBio</h1></a>
        <p id="site-desc"></p>
    </hgroup>
    <nav>
        <ul id="nav-links">
                <li><a href="http://wjidea.github.io/#">Home</a></li>
                <li><a href="http://wjidea.github.io/category/article.html">Blog</a></li>
                <li><a href="http://wjidea.github.io/pages/contact.html">Contact</a></li>
                <li><a href="http://wjidea.github.io/files/researchStatement.pdf">Research</a></li>
                <li><a href="http://wjidea.github.io/files/CV-Wang_noRef.pdf">CV</a></li>
        </ul>
    </nav>
<footer id="site-info">
    <p>
        Proudly powered by <a href="http://getpelican.com/" target="pelican">Pelican</a> and <a href="http://python.org/" target="python">Python</a>. Theme by <a href="https://github.com/hdra/pelican-cait" target="github">hndr</a>.
    </p>
    <p>
    	Maintained by Â© <a href="http://wangjie.esy.es" target="homepage">Jie Wang</a>
    </p>
</footer></header>
    <div id="post-container">
        <ol id="post-list">
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-04-28T00:00:00-04:00" pubdate>
                            Thu 28 April 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/rMaps.html" rel="bookmark"><h1>Making maps in R</h1></a>
                    </header>
                    <section class="post-content">
                        <p>I have been wokring on my population genetics project. One of the main thing is to plot the data in the context of the sampling locations. I have been searching all different kinds of solutions. One of the best for my application was to use the packages: maps, mapplots. </p>
<p>Below ...</p>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-04-26T00:00:00-04:00" pubdate>
                            Tue 26 April 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/pybiomodule.html" rel="bookmark"><h1>Bioinformatics related _Python_ modules</h1></a>
                    </header>
                    <section class="post-content">
                        <ul>
<li><a href="https://bitbucket.org/james_taylor/bx-python/wiki/Home">bx-python</a></li>
<li><a href="https://github.com/chapmanb/bcbb">bcbb</a></li>
<li><a href="https://github.com/chapmanb/bcbio-nextgen">bcbio-nextgen</a></li>
<li><a href="https://bioconda.github.io/">bioconda</a></li>
<li><a href="http://pyvcf.readthedocs.org/">PyVCF</a></li>
<li><a href="http://scikit-learn.org/stable/">SciKit-Learn</a></li>
</ul>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-03-12T00:00:00-05:00" pubdate>
                            Sat 12 March 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/installEgglib.html" rel="bookmark"><h1>Compile Bio++ and install egglib on Ubuntu 14.04 LTS</h1></a>
                    </header>
                    <section class="post-content">
                        <p>I didn't find it difficult to install or compile the egglib module on Linux, when I first come to use egglib in python. However, things changed my mind when I was trying to use the internal function Align.polymorphismBPP(), which is a function that depends on the Bio++ library ...</p>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-03-10T00:00:00-05:00" pubdate>
                            Thu 10 March 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/goodPackages.html" rel="bookmark"><h1>Good and useful bioinfomatics tools</h1></a>
                    </header>
                    <section class="post-content">
                        <p>Here below are the list of softwares I found to be very handy for applied bioinformatics. Most of the tools I listed here are open-source program.</p>
<ul>
<li><a href="http://wasabiapp.org/software/prank/">PRANK</a> - Codon sequence alignment</li>
<li><a href="http://mafft.cbrc.jp/alignment/software/">MAFFT</a> - Multiple sequence alignment, similar performance to <a href="http://www.drive5.com/muscle/downloads.htm">MUSCLE</a></li>
<li><a href="http://egglib.sourceforge.net/">egglib</a> - powerful tools for population genetics and genomics data analysis</li>
<li><a href="http://sanger-pathogens.github.io/Roary/">Roary</a> - tool ...</li></ul>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-03-09T00:00:00-05:00" pubdate>
                            Wed 09 March 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/popSim.html" rel="bookmark"><h1>Population genetics simulation in Python</h1></a>
                    </header>
                    <section class="post-content">
                        <p>A python script writen in OOP style to simulate population genetic dfrit and spontaneous mutation. Microsatellite markers were used as a surrogate to calculate genetic diversity across time/generations.</p>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2016-03-08T00:00:00-05:00" pubdate>
                            Tue 08 March 2016
                        </time>
                        <a href="http://wjidea.github.io/2016/bacpan1.html" rel="bookmark"><h1>Pan-Core Genome Pipeline (Part I)</h1></a>
                    </header>
                    <section class="post-content">
                        <p>Parallel<br />
- <a href="https://www.youtube.com/watch?v=OpaiGYxkSuQ">How to use Parallel (video)</a><br />
- <a href="http://www.gnu.org/software/parallel/parallel_tutorial.html">Tutorial Document</a></p>
<p>FASTQC
`
    #!/bin/bash
    #this file is to run fastqc</p>
<div class="highlight"><pre>#run paralell
#-j indicates run as many jobs as possible
#+0 indicates add 0 job to cpu core(s)
ls *.gz | time parallel -j+0 --eta &#39;fastqc {}&#39;`
</pre></div>


<p>Trimmomatic</p>
<div class="highlight"><pre>TRIM_PATH=&#39;/home/wjidea/apps/Trimmomatic-0 ...</pre></div>
                    </section>
                </article>
            </li>
            <hr/>
            <li>
                <article class="post-entry">
                    <header class="entry-header">
                        <time class="post-time" datetime="2014-08-01T00:00:00-04:00" pubdate>
                            Fri 01 August 2014
                        </time>
                        <a href="http://wjidea.github.io/2014/builtServer.html" rel="bookmark"><h1>Build a bioinformatics lab server with a small budget in 2014 (Part1)</h1></a>
                    </header>
                    <section class="post-content">
                        <p>A friend of mine is going to set up a plant pathology lab soon, and he is planning working with some bioinformatics work over there. I am helping him build a lab server with a reasonable price, which in this case was USD1.5k. I plan to use this opportunity ...</p>
                    </section>
                </article>
            </li>
            <hr/>
        </ol>
        <div class="paginator">
            Page 1 / 1
        </div>

    </div>
        </div>

        <script src="http://wjidea.github.io/theme/js/main.js"></script>
    </body>
</html>